shop/src/main/java/ro/msg/learning/shop/ShopApplication.java

```diff
@@ -2,8 +2,10 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
 
 @SpringBootApplication
+@ComponentScan("ro.msg.learning.shop.controllers")
```

Comment:
@SpringBootApplication already contains the @componentscan annotation with the default value, which means current package and all sub-packages (so all the packages mentioned by you here)
So, its redundant to mention it. 
P.S. this might be helpful: https://www.baeldung.com/spring-boot-annotations